### PR TITLE
Make the example documentation clearer

### PR DESCRIPTION
**Pull request by psiphi75**
Added comments on updating the certs/openssl.cnf for access across the network.
Changed the path of the certificates folder, since then it will just be copy and paste.
Changed the `.` to a `./` because when I copy-n-pasted I missed the `.` because it's not very visible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,10 +50,14 @@ Example certificates are included in the repository for test purposes.
 
 ```sh
 $ cd quinn
-$ cargo run --example server -- --cert ../certs/server.chain --key ../certs/server.rsa .
-$ cargo run --example client -- --ca ../certs/ca.der https://localhost:4433/Cargo.toml
+$ cargo run --example server -- --cert ./certs/server.chain --key ./certs/server.rsa ./
+$ cargo run --example client -- --ca ./certs/ca.der https://localhost:4433/Cargo.toml
```
Comment (Ralith) [inline README.md]: I didn't realize cargo would search workspace members for examples! Please also remove the `cd quinn` line so this is consistent.

File: README.md [hunk 2]
```diff
@@ -50,10 +50,14 @@ Example certificates are included in the repository for test purposes.
...
 ```
 
+The above example will work on localhost and serve the "Cargo.toml" file to the client.  If
```
Comment (Ralith) [inline README.md]: This seems slightly misleading, as the server serves the entire directory supplied to it.

File: README.md [hunk 3]
```diff
@@ -50,10 +50,14 @@ Example certificates are included in the repository for test purposes.
...
+you wish to run across a network you need to update the `certs/openssl.cnf` file and change 
+the `DNS.3` entry to suit the DNS name of the server.
```
Comment (Ralith) [inline README.md]: Wouldn't you need to regenerate the certificates after changing the configuration? Might also want to be clearer that you need to do this if you want to use these certs; the more typical way to run across a network would be to use real certs from e.g. Let's Encrypt.

File: README.md [hunk 4]
```diff
@@ -49,11 +49,15 @@ The example client [currently always verifies][insecure] the server's certificat
...
-$ cargo run --example server -- --cert ../certs/server.chain --key ../certs/server.rsa .
-$ cargo run --example client -- --ca ../certs/ca.der https://localhost:4433/Cargo.toml
+$ cargo run --example server -- --cert ./certs/server.chain --key ./certs/server.rsa ./
+$ cargo run --example client -- --ca ./certs/ca.der https://localhost:4433/Cargo.toml
 ```
 
+In the above example, the server will run on localhost and serve the "." folder to the client.
+The client will request the "Cargo.toml" file.  If you wish to run across a network you need 
+to update the `certs/openssl.cnf` file and change the `DNS.3` entry to suit the DNS name of 
+the server, and then regenerate the certificates using the `certs/generate.sh` script.
```
[the remark this answers is not in the document]
Comment (psiphi75) [reply README.md]: I've updated the description with your previous suggestion.  I haven't tested the `dangerous_configuration` / `--accept-insecure-certs`, so am not really clear on how that works.

[the remark this answers is not in the document]
Comment (Ralith) [reply README.md]: `dangerous_configuration` is a cargo feature flag, which enables the rustls options that allow the default certificate verification procedure to be overriden, the quinn `ClientConfigBuilder::accept_insecure_certs` helper method to thereby disable verification entirely, and the corresponding command line flag in the client example, allowing a client to communicate with servers using invalid or untrusted (e.g. self-signed) certificates.

File: README.md [hunk 5]
```diff
@@ -49,11 +49,20 @@ The example client [currently always verifies][insecure] the server's certificat
...
+$ cargo run --example client -- --ca ./certs/ca.der https://localhost:4433/Cargo.toml
 ```
 
+In the above example, the server will run on localhost and serve the "." folder to the client.
+The client will request the "Cargo.toml" file.  
+
+If you wish to run the client/server across a network for testing only (not production) you 
+need to update the `certs/openssl.cnf` file and change the `DNS.3` entry to suit the DNS name 
+of the server, and then regenerate the certificates using the `certs/generate.sh` script.  This
+method of generating certificates is very limiting. The preferred method when running quinn
```
Comment (Ralith) [inline README.md]:
```suggestion
method of generating certificates is very limiting. The preferred method when running a 
```

File: README.md [hunk 6]
```diff
@@ -49,11 +49,20 @@ The example client [currently always verifies][insecure] the server's certificat
...
 ```
 
+In the above example, the server will run on localhost and serve the "." folder to the client.
+The client will request the "Cargo.toml" file.  
+
+If you wish to run the client/server across a network for testing only (not production) you 
+need to update the `certs/openssl.cnf` file and change the `DNS.3` entry to suit the DNS name 
+of the server, and then regenerate the certificates using the `certs/generate.sh` script.  This
+method of generating certificates is very limiting. The preferred method when running quinn 
+server on a network would be to use real certs from an actual certificate authority, like
```
Comment (Ralith) [inline README.md]:
```suggestion
server on a network is to use real certs from an actual certificate authority, like 
```

File: README.md [hunk 7]
```diff
@@ -49,11 +49,20 @@ The example client [currently always verifies][insecure] the server's certificat
...
-$ cargo run --example server -- --cert ../certs/server.chain --key ../certs/server.rsa .
-$ cargo run --example client -- --ca ../certs/ca.der https://localhost:4433/Cargo.toml
+$ cargo run --example server -- --cert ./certs/server.chain --key ./certs/server.rsa ./
+$ cargo run --example client -- --ca ./certs/ca.der https://localhost:4433/Cargo.toml
 ```
 
+In the above example, the server will run on localhost and serve the "." folder to the client.
+The client will request the "Cargo.toml" file.  
+
+If you wish to run the client/server across a network for testing only (not production) you
```
Comment (Ralith) [inline README.md]:
```suggestion
To run the example client/server across a network you 
```
(and reflow)

**Comment by psiphi75** [comment]
@Ralith, it seems this is ready to merge.  Thanks for your feedback.

**Comment by Ralith** [comment]
This'll need to be rebased and squashed before it can be merged.

**Comment by Ralith** [comment]
Specifically, could you collapse the changes into a single commit on top of current master?

**Comment by psiphi75** [comment]
I'll close this PR and have opened #115.  I hope this is satisfactory.

**Pull request closed without merging**